src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java

```diff
@@ -99,12 +100,32 @@ class CreateGroupPolling extends Polling {
 
             @Override
             public Boolean call() throws Exception {
-                group = securityClient.createGroup((groupName), 201);
+                group = createGroup(groupName);
+
                 // set permissions
                 permissionsObj.changePermissions(group.getId(), CONTENT_NODE, true, true, true, false,
                         false, false, false, 200);
                 return true;
             }
+
+            private Group createGroup(String groupName) throws InterruptedException, ClientException {
+                try {
+                    group = securityClient.createGroup((groupName), 201);
+                } catch (ClientException e) {
+                    if (HttpStatus.SC_CONFLICT == e.getHttpStatusCode()) {
+                        LOG.warn("409 Conflict detected at group {} creation.", groupName);
+                        if (Group.exists(securityClient, groupName)) {
+                            LOG.warn("409 Conflict group {} already exists. Using it.", groupName);
+                            group = new Group(securityClient, groupName);
```

Comment:
Why not `return group` here and get rid of the 2 `else { throw e; }` blocks and just `throw e` at the end of the catch block on L114?